src/Swarm/Game/Step.hs

```diff
@@ -1627,13 +1641,13 @@ execConst c vs s k = do
             missingParentInv = neededParentInv `E.difference` parentInventory
             missingMap =
               M.fromList
-                . filter ((> 0) . snd)
+                . filter (\(n, x) -> x > 0 && (x /= PosInfinity || countByName n parentInventory /= PosInfinity))
                 . map (swap . second (^. entityName))
                 . E.elems
                 $ missingParentInv
 
         -- If we're missing anything, throw an error
-        E.isEmpty missingParentInv
+        null missingMap
```

Comment:
Why did this change from checking `missingParentInv` to checking `missingMap`?  Was this a bug before?

Reply:
I am not sure. But if I hack in the infinity check above, I also need to change this.

The inventory version is likely smart and will ignore 0. See my future comment about the overall data structure.